core/src/main/java/org/jdbi/v3/core/mapper/reflect/ImmutableMapper.java

```diff
@@ -194,41 +166,44 @@ public T map(final ResultSet rs, final StatementContext ctx) throws SQLException
                 "didn't find any matching columns in result set", type));
         }
 
+        Constructor<MethodHandles.Lookup> constructorTemp = null;
+        try {
+            constructorTemp = MethodHandles.Lookup.class.getDeclaredConstructor(Class.class, int.class);
+            constructorTemp.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+        }
+        final Constructor<MethodHandles.Lookup> constructor = constructorTemp;
+
         return (r, c) -> {
-            Object builder = null;
-            try {
-                builder = builderCreationMethod.invoke(null);
-            } catch (IllegalAccessException | InvocationTargetException e) {
-                throw new MappingException("Could not create Builder", e);
-            }
+            final Map<Method, Object> invocations = new HashMap<>();
             for (int i = 0; i < mappers.size(); i++) {
                 RowMapper<?> mapper = mappers.get(i);
                 PropertyDescriptor property = properties.get(i);
-
                 Object value = mapper.map(r, ctx);
-
-                try {
-                    MethodHandles.lookup().unreflect(property.getWriteMethod()).bindTo(builder).invokeWithArguments(value);
-                } catch (IllegalAccessException e) {
-                    throw new MappingException("Setter for the builder should be public", e);
-                } catch (RuntimeException e) {
-                    throw e;
-                } catch (Throwable e) {
-                    throw new MappingException("Immutable build failed with exception", e);
-                }
+                invocations.put(property.getReadMethod(), value);
```

Comment:
It seems like you're trying to roll your own Immutables?

I'd much rather just integrate with what Immutables already does than try to roll our own here--why don't we just spin up a separate `jdbi3-immutables` module?